### PR TITLE
Create slashing.md

### PR DESCRIPTION
**Pull request by rob-solana**
WIP

#### Problem
slashing mechanics and implications are thinly covered in the book

#### Summary of Changes
add a slashing chapter

Fixes #

### CODE REVIEW

File: book/src/slashing.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+Solana implements slashing in 2 regimes: the validator regime and the replicator regime, corresponding to the two rewards regimes.
+During slashing, the network destroys some "penalty percentage" of lamports staked/earned in the rewards-generating accounts.
+
+## Validator regime
+
+Validators cast their votes to a Vote account to which stake has been [passively delegated](stake-delegation-and-rewards.md).
+The Vote program keeps track of rewards for all of the stakes, which allows it to be a central point for slashing.
+The basic mechanism of slashing is:
+
+1. a proof of byzantine behavior is submitted to the Vote program
```
Comment (aeyakovenko) [inline book/src/slashing.md]: @garious would barf on the `byzantine`

File: book/src/slashing.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+Since any entity on the network can submit a rewards redemption instruction to a Stake account, stakes can be 
+slashed at the network's leisure.  The lamports in a Stake or Vote account cannot be reclaimed from those 
+accounts without some access, so the lamports are effectively destroyed if no slashing transactions are 
+ever sent to the Stake program.
+
+### Byzantine behavior (i.e. slashing proofs)
+
+#### Vote transaction as proof
+
+Some proofs of a conflicting vote can be as simple as a Vote transaction, because the Vote account keeps track
```
Comment (aeyakovenko) [inline book/src/slashing.md]: this is a bit hand wavy.  Can we define the actual proofs, and leave the rest as TBD?

File: book/src/slashing.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+Some proofs of byzantine behavior require more data.
+1. conflicting transmission: 2 blobs or 2 shreds that are signed by a validator as leader that have the same slot and index, 
+but differing signatures.  The proof comprises `S(slot, index, hash(data)) + slot, index, hash(data)` for each of the conflicting blobs.
+2. ..
+
+### Slashing's implications on network
+
+#### Consensus
+
+The network removes slashed stakes from the consensus calculations.  A voting strategy of waiting for 2/3rds stake at a lockout
```
Comment (aeyakovenko) [inline book/src/slashing.md]: each validator should remove the slashed stakes.  But this computation is of-chain, so its up to them.

File: book/src/slashing.md (hunk 1 above)
Comment (garious) [reply book/src/slashing.md]: Nah, I've come around to it. Just capitalize it please.

File: book/src/slashing.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+## Validator regime
+
+Validators cast their votes to a Vote account to which stake has been [passively delegated](stake-delegation-and-rewards.md).
+The Vote program keeps track of rewards for all of the stakes, which allows it to be a central point for slashing.
+The basic mechanism of slashing is:
+
+1. a proof of Byzantine behavior is submitted to the Vote program
+2. the Vote program marks the Vote account as "slashed", which means:
+    * no further votes may be submitted to this account
```
Comment (aeyakovenko) [inline book/src/slashing.md]: This punishment is worse then death :). Since validators have to get all their customers to redelegate.

Comment (rob-solana) [reply book/src/slashing.md]: yes.  alternative?

once slashed, the account should probably hang for some amount of time, then be allowed to evaporate

none of the above is implemented

I suppose I can put a placeholder in for `slash()` and figure it out later.

Comment (aeyakovenko) [reply book/src/slashing.md]: The signing key is marked invalid, account is frozen for an epoch?

Comment (rob-solana) [reply book/src/slashing.md]: and slashing (i.e. sending TXs to all the stakes) can only be done while the account is frozen?

Comment (aeyakovenko) [reply book/src/slashing.md]: Are we slashing stakes individually?

Comment (rob-solana) [reply book/src/slashing.md]: we can batch 'em I suppose (limited number of accounts per TX), but someone has to pay for the TXes that mod the stakes

**Pull request closed without merging**